### PR TITLE
zapi: cli for interacting with zqd

### PR DESCRIPTION
**Pull request by mattnibs**
The zapi is command line interface for issuing requests against
the zqd api. Currently the cli allows for:
- creating/delete/viewing spaces
- running searches on spaces
- post pcap files to spaces

This also includes a fully functioning go client for making requests
against the zqd api.

### CODE REVIEW

File: cmd/zapi/cmd/get/get.go [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+	f.StringVar(&c.outputFile, "o", "", "write data to output file")
+	f.BoolVar(&c.verbose, "v", false, "show verbose details")
+	f.BoolVar(&c.reverse, "R", false, "reverse search order (from oldest to newest)")
+	f.BoolVar(&c.stats, "S", false, "display search stats on stderr")
+	f.BoolVar(&c.warnings, "W", false, "display warnings on stderr")
+	f.BoolVar(&c.ShowTypes, "T", false, "display field types in text output")
+	f.BoolVar(&c.ShowFields, "F", false, "display field names in text output")
+	f.BoolVar(&c.EpochDates, "E", false, "display epoch timestamps in text output")
+	f.BoolVar(&c.nocache, "C", false, "bypass the aggregations cache")
+	f.BoolVar(&c.noindex, "X", false, "search brute force without search index")
```
Comment (henridf) [inline cmd/zapi/cmd/get/get.go]: is this relevant here?

File: cmd/zapi/cmd/get/get.go [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+	f.BoolVar(&c.wire, "w", false, "dump whats on the wire")
```
Comment (henridf) [inline cmd/zapi/cmd/get/get.go]: s/whats/what's/

File: cmd/zapi/cmd/get/get.go [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+	f.StringVar(&c.dir, "d", "", "directory for output data files")
+	f.StringVar(&c.outputFile, "o", "", "write data to output file")
+	f.BoolVar(&c.verbose, "v", false, "show verbose details")
+	f.BoolVar(&c.reverse, "R", false, "reverse search order (from oldest to newest)")
+	f.BoolVar(&c.stats, "S", false, "display search stats on stderr")
+	f.BoolVar(&c.warnings, "W", false, "display warnings on stderr")
+	f.BoolVar(&c.ShowTypes, "T", false, "display field types in text output")
+	f.BoolVar(&c.ShowFields, "F", false, "display field names in text output")
+	f.BoolVar(&c.EpochDates, "E", false, "display epoch timestamps in text output")
+	f.BoolVar(&c.nocache, "C", false, "bypass the aggregations cache")
```
Comment (henridf) [inline cmd/zapi/cmd/get/get.go]: nocache?

File: cmd/zapi/cmd/get/get.go (hunk 1 above)
Comment (mattnibs) [reply cmd/zapi/cmd/get/get.go]: copy pasta. Thanks.

File: cmd/zapi/cmd/get/get.go (hunk 3 above)
Comment (mattnibs) [reply cmd/zapi/cmd/get/get.go]: Copy pasta. Will remove.

**Comment by mattnibs** [comment]
@alfred-landrum sure

**Comment by mattnibs** [comment]
Will revive in smaller, more palatable chunks.

**Pull request closed without merging**